Themes/default/scripts/script.js

```diff
@@ -34,6 +34,19 @@ if (!('getElementsByClassName' in document))
 	}
 }
 
+window.addEventListener('DOMContentLoaded', function() {
+	for (const elem of document.querySelectorAll('.bbc_inline_spoiler')) {
+		elem.addEventListener('click', toggleSpoiler);
+	}
+});
+
+// Shows or hides the content of a spoiler tag.
+function toggleSpoiler() {
+	const button = this.firstElementChild;
+	button.nextElementSibling.style.visibility = (button.nextElementSibling.style.visibility === 'visible' ? 'hidden' : 'visible');
+	this.style.backgroundColor = (this.style.backgroundColor === 'transparent' ? 'currentcolor' : 'transparent');
```

Comment:
Can both of these colours be moved somewhere where they can be customised (eg a custom property)? `currentcolor` won't necessarily be desirable as the spoiler colour, especially in a dark theme. As for using `transparent` as the 'shown' background, I'd suggest using a background that is slightly visible to indicate that it came from a spoiler, especially since you can toggle back and forth.

Reply:
I think you misunderstand what `currentcolor` will do here. It forces the background to match the text colour. So if the text is a light colour on a dark background, the background colour applied to the hidden spoiler will also be the same light colour. I chose `currentcolor` precisely because it is the best way to both light and dark themes with the same CSS.

It would be possible to apply some non-transparent background to a revealed spoiler, yes, but I don't see the benefit to that. Using `transparent` both guarantees compatibility with all themes and keeps the JavaScript used to toggle between states simple. It would be possible on both counts to change that, but it would be more complex and thus easier for a custom theme to mess up.

Reply:
Still, I suppose I could change this to toggle a class instead.

Reply:
What about something like this?
```js
function toggleSpoiler() {
    const button = this.firstElementChild;
    const showing = button.nextElementSibling.style.visibility === 'visible';
    button.nextElementSibling.style.visibility = (showing ? 'hidden' : 'visible');
    this.style.backgroundColor = (showing ? 'var(--bbc-spoilerinline-show-color, currentcolor)' : 'var(--bbc-spoilerinline-hide-color, transparent)');
}
```

Reply:
[f82d5e7](https://github.com/SimpleMachines/SMF/pull/8593/commits/f82d5e76c97c3d5eaa1b529312a942f7671d3b86) is simpler.